celements-model/src/main/java/com/celements/model/object/AbstractObjectEditor.java

```diff
@@ -123,23 +116,39 @@ public O apply(ClassIdentity classId) {
 
     @Override
     public boolean apply(O obj) {
-      boolean success = getBridge().deleteObject(doc, obj);
+      boolean success = getBridge().deleteObject(getDocument(), obj);
       LOGGER.info("{} deleted object {} for {}: {}", AbstractObjectEditor.this,
           getBridge().getObjectNumber(obj), getBridge().getObjectClass(obj), success);
       return success;
     }
   }
 
-  @Override
-  public String toString() {
-    return this.getClass().getSimpleName() + " [doc=" + getBridge().getDocRef(doc) + ", query="
-        + getQuery() + "]";
-  }
-
   @Override
   public abstract AbstractObjectFetcher<?, D, O> fetch();
 
   @Override
-  protected abstract @NotNull ObjectBridge<D, O> getBridge();
+  public <T> FieldEditor<T> editField(final ClassField<T> field) {
+    final Iterable<O> objects = fetch().filter(field.getClassDef()).iter();
+    return new FieldEditor<T>() {
+
+      @Override
+      public boolean first(@Nullable T value) {

```

Comment:
could we add a `public boolean number(@Nullable T value, int objectNr)` to set a value based on the object number. It's not the most common use case, but would be useful from time to time.

Reply:
The FieldFetcher/-Editor simply accept a list objects, they may have been pre-filtered by the ObjectFetcher/-Editor already. Thus your suggested use case is already possible with:

`Optional<T> value = ObjectFetcher.on(doc).filter(classRef).filter(objNb).fetchField(field).first();`
`boolean changed = ObjectEditor.on(doc).filter(classRef).filter(objNb).editField(field).first(value);`